### DISCUSSION TITLE
用了onlyoffice可在线不显示是为啥？

### DISCUSSION THREAD

**Discussion by monobo1989** [AlistGo/alist#6715] [Q&A]
1.按照下图安装了onlyoffice:
![微信图片_20240708072445](https://github.com/alist-org/alist/assets/73515609/66b939a6-19a6-4629-98d7-311dc33185e6)

2.新建了view文件，路径为：/opt/onlyoffice   view文件内容如图：
![微信图片_20240708070536](https://github.com/alist-org/alist/assets/73515609/cd215f55-c53b-4e5f-9d41-cc7ff9f8884c)

3.nginx安装好后，vim /etc/nginx/conf.d/onlyoffice.conf，配置如图：
![微信图片_20240708071041](https://github.com/alist-org/alist/assets/73515609/741795a5-600b-4eb3-9bd9-e45901c17dc3)

4.alist预览配置增加了图中内容： 
![微信图片_20240708071243](https://github.com/alist-org/alist/assets/73515609/089df8e8-e2d1-4115-989e-d6019d437811)

5.可是最终无法预览，如图：
![微信图片_20240708071359](https://github.com/alist-org/alist/assets/73515609/72bc550f-0582-42a3-ac0b-4e9e7cb8285e)

请问各位，是哪里出问题了呢？谢谢啊！！

**Comment by bigwatermelon8** [comment]
这个我折腾好就才搞出来，主要是view.html文件的问题，不是找不到，就是配置错误。你可以手动访问一下view.html文件，看下有没有报错，正常情况下是会有个弹窗出来才正常的

**Comment by superkeyor** [comment]
附上我的命令，直接运行即可
基本逻辑：
1 安装onlyoffice
2 构建view.html,复制到api.js同文件夹，便于外部调用
3 修改onlyoffice配置，如果是局域网内部使用的话
4 重启onlyoffice container！
5 在alist中@manage/settings/preview配置
your_intranet_ip是host的ip，62455是在docker中map的port，可以自己指定不同的值
{
  "pdf,doc,docx,xls,xlsx,ppt,pptx": {
    "OnlyOffice":"http://your_intranet_ip:62455/web-apps/apps/api/documents/view.html?src=$e_url",
    "Microsoft":"https://view.officeapps.live.com/op/view.aspx?src=$e_url",
    "Google":"https://docs.google.com/gview?url=$e_url&embedded=true"
  },
  "pdf": {
    "PDF.js":"https://alist-org.github.io/pdf.js/web/viewer.html?file=$e_url"
  },
  "epub": {
    "EPUB.js":"https://alist-org.github.io/static/epub.js/viewer.html?url=$e_url"
  }
}
```

sudo docker run -i -t -d -p 62455:80 --name myonlyoffice -e JWT_ENABLED=false --restart=always onlyoffice/documentserver:7.2

sudo docker exec myonlyoffice bash -c 'cat << "EOF" > /var/www/onlyoffice/documentserver/web-apps/apps/api/documents/view.html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>OnlyOffice Viewer</title>
</head>

<body>
    <div id="placeholder"></div>
    <script type="text/javascript" src="/web-apps/apps/api/documents/api.js"></script>
    <script>
        function getQueryParamValue(name) {
            const searchParams = new URLSearchParams(window.location.search);
            return searchParams.get(name);
        }

        const url = decodeURIComponent(getQueryParamValue("src"));
        const fileName = url.substring(url.lastIndexOf("/") + 1, url.lastIndexOf("?") != -1 ? url.lastIndexOf("?") : url.length);
        const fileExtension = fileName.split(".").pop();
        const docEditor = new DocsAPI.DocEditor("placeholder", {
            "document": {
                "fileType": fileExtension,
                "permissions": {
                    "edit": false,
                    "comment": true,
                    "download": true,
                    "print": true,
                    "fillForms": true,
                },
                "title": fileName,
                "url": url,
            },
            "editorConfig": {
                "lang": "en-US",
                "mode": "view",
            },
            "height": "1080px",
            "type": "desktop",
        });
    </script>
</body>
</html>
EOF'

# 关闭ssl，允许使用IP
# https://blog.pwolf.site/2023/09/15/%E5%B1%80%E5%9F%9F%E7%BD%91%E5%86%85%E9%83%A8%E7%BD%B2alist%E5%92%8Conlyoffice/
sudo docker exec myonlyoffice bash -c "
# Modify the configuration file
sed -i 's/\"rejectUnauthorized\": true/\"rejectUnauthorized\": false/' /etc/onlyoffice/documentserver/default.json
sed -i 's/\"allowPrivateIPAddress\": false/\"allowPrivateIPAddress\": true/' /etc/onlyoffice/documentserver/default.json
sed -i 's/\"allowMetaIPAddress\": false/\"allowMetaIPAddress\": true/' /etc/onlyoffice/documentserver/default.json
"

sudo docker restart myonlyoffice


```

**Reply by superkeyor** [reply]
我把命令集成到image，更加简单了，省去上面1-4步，直接拉docker
```
    myonlyoffice:
        image: superkeyor/onlyoffice
        container_name: myonlyoffice
        ports:
            - "62455:80"
        restart: always
```
具体修改可以参见https://github.com/superkeyor/onlyoffice/blob/main/Dockerfile
貌似default.json的几个参数可以通过env修改：https://github.com/ONLYOFFICE/Docker-DocumentServer?tab=readme-ov-file#available-configuration-parameters
